client/my-sites/themes/themes-magic-search-card/index.jsx

```diff
@@ -0,0 +1,110 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import debounce from 'lodash/debounce';
+
+/**
+ * Internal dependencies
+ */
+import Search from 'components/search';
+import SegmentedControl from 'components/segmented-control';
+import { trackClick } from '../helpers';
+import config from 'config';
+import { isMobile } from 'lib/viewport';
+
+const ThemesMagicSearchCard = React.createClass( {
+	propTypes: {
+		tier: React.PropTypes.string,
+		select: React.PropTypes.func.isRequired,
+		site: React.PropTypes.oneOfType( [
+			React.PropTypes.object,
+			React.PropTypes.bool
+		] ).isRequired,
+		onSearch: React.PropTypes.func.isRequired,
+		search: React.PropTypes.string
+	},
+
+	trackClick: trackClick.bind( null, 'search bar' ),
+
+	componentWillMount() {
+		this.onResize = debounce( () => {
+			this.setState( { isMobile: isMobile() } );
+		}, 250 );
+	},
+
+	componentDidMount() {
+		window.addEventListener( 'resize', this.onResize );
+	},
+
+	componentWillUnmount() {
+		window.removeEventListener( 'resize', this.onResize );
+	},
+
+	getInitialState() {
+		return {
+			isMobile: isMobile(),
+			searchIsOpen: false
+		};
+	},
+
+	getDefaultProps() {
+		return { tier: 'all' };
+	},
+
+	onSearchOpen() {
+		this.setState( { searchIsOpen: true } );
+	},
+
+	onSearchClose() {
+		this.setState( { searchIsOpen: false } );
+	},
+
+	onBlur() {
+		if ( this.state.isMobile ) {
+			this.setState( { searchIsOpen: false } );
+		}
+	},
+
+	render() {
+		const isJetpack = this.props.site && this.props.site.jetpack;
```

Comment:
![19hagt](https://cloud.githubusercontent.com/assets/3392497/17966548/72456370-6ac2-11e6-893c-2a7f0701784c.jpg)

😜 
Multiple `const` declarations are OK, but if they are, for example, in different scopes. We discussed this on Slack and the current code guidelines have been [loosened a lot](https://github.com/Automattic/wp-calypso/commit/692d611e48a9c79ba33a8bc6a308db2b793d60f9) so it's not clear if this is allowed or not. IMHO, this convention looks a bit out of place in Calypso - @folletto can you weigh in here? 🙇